File: backend/schemas/oauth.py
from __future__ import annotations

from .common import FlexibleSchema


class OAuthStartRequest(FlexibleSchema):
    profileName: str | None = None
    region: str | None = "cn"
    tenantId: str | None = "common"
    clientId: str
    clientSecret: str | None = ""
    redirectUri: str | None = "http://127.0.0.1:17651/api/oauth/callback"
    scopes: str | None = ""
    driveId: str | None = ""
    siteId: str | None = ""
    siteHostname: str | None = ""
    sitePath: str | None = ""
    libraryName: str | None = ""
    rootPath: str | None = "/media"


class OAuthStartResponse(FlexibleSchema):
    ok: bool = True
    authorizationUrl: str
    state: str
    redirectUri: str
    scopes: str

File: frontend/apps/web-antd/src/api/oauth.ts
import { post, request } from './request';

export interface OAuthStartBody {
  profileName?: string;
  region?: string;
  tenantId?: string;
  clientId?: string;
  clientSecret?: string;
  redirectUri?: string;
  scopes?: string;
  driveId?: string;
  siteId?: string;
  siteHostname?: string;
  sitePath?: string;
  libraryName?: string;
  rootPath?: string;
}

export interface OAuthStartResponse {
  ok: true;
  authorizationUrl: string;
  state: string;
  redirectUri: string;
  scopes: string;
}

export const oauthApi = {
  start: (body: OAuthStartBody) => post<OAuthStartResponse>('/api/oauth/start', body),
  result: (state: string) => request<{ ok: true; result: Record<string, unknown> | null }>(`/api/oauth/result/${state}`)
};
